src/Main.java

```diff
@@ -12,14 +13,205 @@ public static void main(String[] args) {
+        System.out.println("\nСотрудник с минимальной зарплатой в отделе #" + departmentNumber + ":");
+        System.out.println(" " + minSalaryEmployee);
+
+        Employee maxSalaryEmployee = findMaxSalaryInDepartment(employees, departmentNumber);
+        System.out.println("\nСотрудник с маскимальной зарплатой в отделе #" + departmentNumber + ":");
+        System.out.println(" " + maxSalaryEmployee);
+
+        double totalSalaryInDepartment = findAllSalaryInDepartment(employees, departmentNumber);
+        System.out.println("\nСумма затрат на зарплату в отделе #" + departmentNumber + ":");
+        System.out.println(" " + totalSalaryInDepartment);
+
+        double midlSalaryInDepartment = findMidlSalaryInDepartment(employees, departmentNumber);
+        System.out.println("\nСредняя зарплата в отделе #" + departmentNumber + ":");
+        System.out.println(" " + midlSalaryInDepartment);
+
+        indexSalaryInDepartment(employees, departmentNumber, 20.0);
+
+        System.out.println("\nСотрудники в отделе #" + departmentNumber + ":");
+        printEmployeesInDepartment(employees, departmentNumber);
+
+        //Для проверки зп
+        double salaryThreshold = 120000.0;
+        System.out.println("\nСотрудники с зарплатой меньше " + salaryThreshold + ":");
+        printEmployeesWithSalaryLessThan(employees, salaryThreshold);
+
+        System.out.println("Сотрудники с зарплатой больше или равной " + salaryThreshold + ":");
+        printEmployeesWithSalaryGreaterThanOrEqual(employees, salaryThreshold);
+
+
+//        printAllName(employees);
+    }
+
+    public static void printEmployeeDetails(Employee[] employees) {
+        for (Employee employee : employees) {
             if (employee != null) {
                 System.out.println(employee.toString());
             }
-        System.out.println("Сумма всех затрат на зрплаты: \n" + Employee.findSumAllSalary(employees) + "\n");
-        System.out.println("Сотрудник с самой низкой зарплатой: \n" + Employee.findMinSalary(employees));
-        System.out.println("Сотрудник с самой высокой зарплатой: \n" + Employee.findMaxSalary(employees));
-        System.out.println("Средняя зарплата среди сотрудников: \n" + Employee.findMidlSalary(employees));
-        Employee.printAllName(employees);
+        }
+    }
+
+    public static double findSumAllSalary(Employee[] employees) {
+        double total = 0.0;
+        for (Employee employee : employees) {
+            if (employee != null) {
+                total += employee.getSalary();
+            }
+        }
+        return total;
+    }
+
+    public static Employee findMinSalary(Employee[] employees) {
+        Employee minSalaryEmployee = null;
+        for (Employee employee : employees) {
+            if (employee != null) {
+                if (minSalaryEmployee == null || employee.getSalary() < minSalaryEmployee.getSalary()) {
+                    minSalaryEmployee = employee;
+                }
+            }
```

Comment:
```suggestion
            if (employee != null && (minSalaryEmployee == null || employee.getSalary() < minSalaryEmployee.getSalary())) {
                minSalaryEmployee = employee;
            }
```
вложенные if лучше объединить в один, убрав вложенность

Reply:
понял